app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+  has_many :likes
+
+  def update_counter
+    user.increment!(:post_counter)
+  end
```

Comment:
- [ ] Great work on defining the methods for your post model! 😊 To enhance the maintainability and organization of your code, I recommend considering the use of [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html), specifically `after_save`, to automatically trigger the `update_counter` method each time a new post is saved to the database. This approach will help you keep your code clean and ensure that the counter is always up to date without manual intervention.
   - Additionally, to encapsulate the `update_counter` method properly and prevent accidental external calls, consider marking it as private. This encapsulation will improve code readability and adherence to best practices.

Happy coding! 👍👨‍💻🚀